pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java

```diff
@@ -570,6 +572,81 @@ public static String[] split(String input, String delimiter, int limit) {
     return StringUtils.splitByWholeSeparator(input, delimiter, limit);
   }
 
+  /**
+   * @param input an input string for prefix strings generations.
+   * @param length the max length of the prefix strings for the string.
+   * @param regexChar the character for regex matching to be added to prefix strings generated. e.g. '^'
+   * @return generate an array of prefix strings of the string that are shorter than the specified length.
+   */
+  @ScalarFunction
+  public static String[] prefix(String input, int length, String regexChar) {
+    ObjectSet<String> prefixSet = new ObjectLinkedOpenHashSet<>();
+    for (int prefixLength = 1; prefixLength <= length && prefixLength <= input.length(); prefixLength++) {
+      if (regexChar != null) {
+        prefixSet.add(regexChar + input.substring(0, prefixLength));
+      } else {
+        prefixSet.add(input.substring(0, prefixLength));
+      }
+    }
+    return prefixSet.toArray(new String[0]);
+  }
+
+  /**
+   * @param input an input string for suffix strings generations.
+   * @param length the max length of the suffix strings for the string.
+   * @param regexChar the character for regex matching to be added to suffix strings generated. e.g. '$'
+   * @return generate an array of suffix strings of the string that are shorter than the specified length.
+   */
+  @ScalarFunction
+  public static String[] suffix(String input, int length, String regexChar) {
+    ObjectSet<String> suffixSet = new ObjectLinkedOpenHashSet<>();
+    for (int suffixLength = 1; suffixLength <= length && suffixLength <= input.length(); suffixLength++) {
+      if (regexChar != null) {
+        suffixSet.add(input.substring(input.length() - suffixLength) + regexChar);
+      } else {
+        suffixSet.add(input.substring(input.length() - suffixLength));
+      }
+    }
+    return suffixSet.toArray(new String[0]);
+  }
+
+  /**
+   * @param input an input string for ngram generations.
+   * @param length the max length of the ngram for the string.
+   * @return generate an array of ngram of the string that length are exactly matching the specified length.
+   */
+  @ScalarFunction
+  public static String[] ngram(String input, int length) {
```

Comment:
Suggest renaming this to `ngrams` to be consistent with CH: https://clickhouse.com/docs/en/sql-reference/functions/splitting-merging-functions#ngrams

Reply:
thanks for the review, this sounds good to me. i will rename it